### ISSUE TITLE
Serving Galaxy with a url prefix broken in 19.01

### ISSUE THREAD

**Issue by abretaud** [galaxyproject/galaxy#7474]
Hello!
As I was testing 19.01 version of the docker image I find out that running Galaxy with an url prefix seems to be broken.

Here's a minimal test setup, using the PROXY_PREFIX option of the docker image (which worked fine with 18.09) : https://github.com/abretaud/galaxy_proxy

`docker-compose up -d` then go to http://localhost:8700/foo/bar/: 
-  some assets get a 404 error, e.g. it tries to download http://localhost:8700/static/scripts/bundled/assets/af7ae505a9eed503f8b8e6982036873e.woff2 instead of http://localhost:8700/foo/bar/static/scripts/bundled/assets/af7ae505a9eed503f8b8e6982036873e.woff2
- some pages don't work, e.g. go to data libraries -> it makes an xhr request to http://localhost:8700/api/libraries?deleted=false instead of http://localhost:8700/foo/bar/api/libraries?deleted=false

I suspect the first thing is related to this line https://github.com/galaxyproject/galaxy/blob/release_19.01/client/webpack.config.js#L100 but I'm not a webpack expert
Any help?

**Comment by mvdbeek** [comment]
Can you check whether this is synced up with https://docs.galaxyproject.org/en/master/admin/nginx.html?highlight=proxy%20prefix#serving-galaxy-at-a-url-prefix ?
Especially
```
    mount: /galaxy=galaxy.webapps.galaxy.buildapp:uwsgi_app()
    manage-script-name: true
```

**Comment by abretaud** [comment]
Yep it is:
```
uwsgi:
  socket: unix:///srv/galaxy/var/uwsgi.sock
  mount: /foo/bar/=galaxy.webapps.galaxy.buildapp:uwsgi_app()
  manage-script-name: true
```
And here's the nginx conf in the container
```
$ cat /etc/nginx/conf.d/uwsgi.conf 
# pass to uWSGI by default
location / {
    uwsgi_pass 127.0.0.1:4001;
    include uwsgi_params;
    uwsgi_param SCRIPT_NAME /foo/bar/;
    # forward the remote_user header in case it is set by a previous proxy
    uwsgi_param HTTP_REMOTE_USER $remote_user if_not_empty;
}
```

**Comment by mvdbeek** [comment]
I'm not sure you'll want to have the trailing slash in there, but I don't think that's the issue.
In the nginx config I think you also want to change location to `/foor/bar` instead of `/`, and the same for the static files.

**Comment by shiltemann** [comment]
I've noticed similar issues (since 18.09 release)

see also: https://github.com/galaxyproject/galaxy/issues/7162

**Comment by mvdbeek** [comment]
Hmm, most of this works for me, but upon browsing around I noticed that the workflow menu doesn't load from the welcome page, though it does from the multi data view (which works fine).
fwiw that's my dev config:
```
uwsgi:
  http: 127.0.0.1:8080
  buffer-size: 16384
  processes: 1
  threads: 4
  offload-threads: 2
  master: false
  virtualenv: .venv
  pythonpath: lib
  mount: /galaxy=galaxy.webapps.galaxy.buildapp:uwsgi_app()
  manage-script-name: true
  thunder-lock: false
  die-on-term: true
  hook-master-start: unix_signal:2 gracefully_kill_them_all
  hook-master-start: unix_signal:15 gracefully_kill_them_all
  py-call-osafterfork: true
  enable-threads: true
galaxy:
  cookie_path: '/galaxy'
  database_connection: 'postgresql://localhost/galaxy'
  conda_prefix: /Users/mvandenb/miniconda3
```

**Comment by abretaud** [comment]
@mvdbeek what is your conf for `static_enabled`? It's disabled in the docker image, which is intended I think
Anyway, I think I have a fix for the css/fonts problem, I'm looking into the problem of library, more news soon hopefully

**Comment by dannon** [comment]
@abretaud How are you addressing the css/fonts problem?  I tried addressing this on the client side and having it be automatic based on Galaxy's configuration (using https://webpack.js.org/configuration/output/#outputpublicpath at runtime) but I haven't got it quite working yet.  This and the routing issues noted above are two separate problems.

**Comment by abretaud** [comment]
So https://github.com/galaxyproject/galaxy/pull/7481 seems to fix the publicPath problem

For the routing issue, for some reason `links` is empty at https://github.com/galaxyproject/galaxy/blob/release_19.01/client/galaxy/scripts/onload/loadConfig.js#L22 when browsing to http://localhost:8700/foo/bar/library/list

**Comment by sveinugu** [comment]
We have also discovered these issues a while ago, and have been holding off updating to 18.09 because of them. Will #7481 and any fix to the routing issue be backported into release_18.09, or do we need to jump to 19.01 (starting a new test-cycle with potentially other issues to be fixed)?

**Comment by sveinugu** [comment]
Update: seems #7162 is the issue we have been struggling with in 18.09. The current issue might be new for 19.01. So a backport of any combined fixes are probably out of the question, then, I assume?

**Comment by dannon** [comment]
@sveinugu I'm going to merge the changes addressing this in particular and then will follow up on the other issues, creating one backport PR to address them.

**Comment by dannon** [comment]
Fixed in #7481 (though there are still other open issues for similar url prefix errors, being addressed elsewhere)
